### PR TITLE
[https://nvbugs/6771102][fix] Support Qwen3.5 global FP8 checkpoints

### PR DESCRIPTION
**Pull request by 2ez4bz**
## Dev Engineer Review

- Supports global `FP8` and `MIXED_PRECISION` linear-attention checkpoints.
- Routes FP8 QKV/Z projections through the fused `in_proj_qkvz` shared-scale path.
- Preserves BF16 A/B projections and excluded-module behavior.
- No public API changes are shown.
- Review finding counts are unavailable because no current findings were supplied.

## QA Engineer Review

- Modified `tests/unittest/_torch/models/checkpoints/hf/test_qwen3_5_weight_mapper.py`.
- Strengthened the excluded-module BF16 test with value checks for packed QKV/Z rows.
- Existing tests cover ModelOpt FP8 layout handling, `weight_scale` application, FP8 packing, BF16 fallback, and synthesized-scale removal.
- No exact matching CI or manual-QA test-list entry was found.
- Coverage verdict: `sufficient` for the targeted mapper transformation. End-to-end checkpoint loading is not covered.

## Per-File QA Perspective

- `tensorrt_llm/_torch/models/checkpoints/hf/qwen3_5_weight_mapper.py`: Verify global FP8 and mixed-precision QKV/Z paths, shared-scale propagation, BF16 A/B handling, and excluded-module behavior.
- `tests/unittest/_torch/models/checkpoints/hf/test_qwen3_5_weight_mapper.py`: Verifies ModelOpt FP8 packing, scale application, packed BF16 values, and fallback behavior. It is not listed in the discovered `test-db/` or `qa/` lists.

## Description

* Why?

ModelOpt global FP8 checkpoints store scalar per-tensor scales for split Qwen3.5 linear-attention projections. The mapper passed those scalars into projection packing, causing an IndexError before the model loaded.

* What?

Normalize global FP8 QKV/Z projections through the existing fused shared-scale path while preserving unquantized A/B projections and module exclusions. Add coverage for the ModelOpt checkpoint layout.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/models/checkpoints/hf/qwen3_5_weight_mapper.py [hunk 1]
```diff
@@ -365,7 +365,15 @@ def _requantize_linear_attn_fp8_qkvz(self, weights: dict) -> dict:
         drop_keys = set()
         for prefix, projs in grouped_names.items():
             fused_cfg = qcd.get(f"{prefix}.in_proj_qkvz")
-            if fused_cfg is None or fused_cfg.quant_algo != QuantAlgo.FP8:
+            globally_quantized = (
+                self.config.quant_config.quant_algo == QuantAlgo.FP8
+                and not self.config.quant_config.is_module_excluded_from_quantization(
+                    f"{prefix}.in_proj_qkvz"
+                )
+            )
```
[the remark this answers is not in the document]
Comment (2ez4bz) [reply tensorrt_llm/_torch/models/checkpoints/hf/qwen3_5_weight_mapper.py:373]: Done.

File: tests/unittest/_torch/models/checkpoints/hf/test_qwen3_5_weight_mapper.py [hunk 2]
```diff
@@ -139,3 +140,28 @@ def test_fp8_rowwise_full() -> None:
...
+    for name in ("in_proj_b", "in_proj_a"):
+        weights[f"{checkpoint_prefix}.{name}.weight"] = _bf16(_BA_ROWS)
+
+    mapper = _make_mapper(QuantAlgo.FP8)
+    out = mapper.preprocess_weights(weights)
+
+    assert out[f"{_ATTN_PREFIX}.in_proj_qkvz.weight"].shape == (_PACKED_QKVZ_ROWS, _HIDDEN)
+    assert out[f"{_ATTN_PREFIX}.in_proj_qkvz.weight"].dtype == torch.float8_e4m3fn
+    torch.testing.assert_close(out[f"{_ATTN_PREFIX}.in_proj_qkvz.weight_scale"], torch.tensor(4.0))
+    torch.testing.assert_close(out[f"{_ATTN_PREFIX}.in_proj_qkvz.input_scale"], torch.tensor(5.0))
```
[the remark this answers is not in the document]
Comment (2ez4bz) [reply tests/unittest/_torch/models/checkpoints/hf/test_qwen3_5_weight_mapper.py:179]: Done.

**Review by Wanli-Jiang — APPROVED** [review]
